We should eslint the scripts

`eslint` would probably have caught the mistake of creating a function called `package`. 
It would have avoided https://github.com/mdn/stumptown-content/issues/109

We shouldn't use `eslint` to worry itself about code formatting. We'll [use prettier](https://github.com/mdn/stumptown-content/pull/28) for that once the ADR settles.

> eslint would probably have caught the mistake of creating a function called package.
It would have avoided #109

 55:16  error  Parsing error: The keyword 'package' is reserved 😄

--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -1,0 +1,16 @@
+module.exports = {
+  "env": {
+    "es6": true,
+    "node": true
+  },
+  "overrides": [
+    {
+      "files": [ "scripts/**/*.js" ],
+      "extends": "eslint:recommended",
+      "parserOptions": {
+        "ecmaVersion": 2018,
+        "sourceType": "module"
+      }
+    }
+  ]
+};

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,6 +7,9 @@ branches:
 
 script:
   - npm run test
+  - npm run lint-js
+  - npm run lint-md
+  - npm run spell-md
   # Note, in a near future we're going to want to build-json for *everything*
   # and not just the 'html' section.
   - npm run build-json html

--- a/.travis.yml
+++ b/.travis.yml
@@ -6,7 +6,6 @@ branches:
     - master
 
 script:
-  - npm run test
   - npm run lint-js
   - npm run lint-md
   - npm run spell-md

--- a/package.json
+++ b/package.json
@@ -16,10 +16,11 @@
         "pretty": "prettier --check scripts/**/*.js",
         "prettify": "prettier --write scripts/**/*.js",
         "lint-md": "remark -q -f content",
+        "lint": "eslint .",
         "build-json": "node scripts/build-json/build-json.js",
         "scrape-mdn": "node scripts/scraper/scrape-mdn.js",
         "spell-md": "mdspell -a -n -r -x --en-us 'content/**/!(*contributors).md'",
-        "test": "npm run lint-md && npm run spell-md"
+        "test": "npm run lint && npm run lint-md && npm run spell-md"
     },
     "dependencies": {
         "gray-matter": "4.0.2",

--- a/package.json
+++ b/package.json
@@ -20,7 +20,7 @@
         "build-json": "node scripts/build-json/build-json.js",
         "scrape-mdn": "node scripts/scraper/scrape-mdn.js",
         "spell-md": "mdspell -a -n -r -x --en-us 'content/**/!(*contributors).md'",
-        "test": ""
+        "test": "npm run lint-js && npm run lint-md && npm run spell-md"
     },
     "dependencies": {
         "gray-matter": "4.0.2",

--- a/package.json
+++ b/package.json
@@ -16,11 +16,11 @@
         "pretty": "prettier --check scripts/**/*.js",
         "prettify": "prettier --write scripts/**/*.js",
         "lint-md": "remark -q -f content",
-        "lint": "eslint .",
+        "lint-js": "eslint .",
         "build-json": "node scripts/build-json/build-json.js",
         "scrape-mdn": "node scripts/scraper/scrape-mdn.js",
         "spell-md": "mdspell -a -n -r -x --en-us 'content/**/!(*contributors).md'",
-        "test": "npm run lint && npm run lint-md && npm run spell-md"
+        "test": ""
     },
     "dependencies": {
         "gray-matter": "4.0.2",
@@ -57,6 +57,7 @@
         ]
     },
     "devDependencies": {
+        "eslint": "^6.2.1",
         "prettier": "1.18.2"
     },
     "prettier": {

--- a/scripts/build-json/build-guide-page-json.js
+++ b/scripts/build-json/build-guide-page-json.js
@@ -12,7 +12,6 @@ async function processDirective(elementPath, directive) {
                 type: 'example',
                 content: await examples.packageExample(path.join(elementPath, directiveComponents[1]))
             }
-            break;
         default:
             throw ('Unsupported guide directive');
     }
@@ -23,7 +22,7 @@ async function processProse(elementPath, proseMD) {
         type: 'prose',
         content: await markdown.markdownToHTML(proseMD)
     }
-};
+}
 
 async function buildGuideContentJSON(elementPath, data, content) {
     const result = [];
@@ -35,7 +34,7 @@ async function buildGuideContentJSON(elementPath, data, content) {
         } else {
              result.push(await processProse(elementPath, section));
         }
-    };
+    }
     return result;
 }
 

--- a/scripts/build-json/build-page-json.js
+++ b/scripts/build-json/build-page-json.js
@@ -20,7 +20,7 @@ function writeToFile(json, elementPath) {
       fs.mkdirSync(destDir, { recursive: true });
   }
   fs.writeFileSync(dest, `${JSON.stringify(json, null, 2)}`);
-};
+}
 
 async function processMetaIngredient(elementPath, ingredientName, data) {
     switch (ingredientName) {
@@ -35,15 +35,16 @@ async function processMetaIngredient(elementPath, ingredientName, data) {
             } else {
                 return [];
             }
-        case 'examples':
-            const examplesPaths = data.examples.map(relativePath => path.join(elementPath, relativePath));
-            return await packageExamples(examplesPaths);
+        case 'examples': {
+          const examplesPaths = data.examples.map(relativePath => path.join(elementPath, relativePath));
+          return await packageExamples(examplesPaths);
+        }
         case 'info_box':
             // TODO: implement packaging for info boxes
             // See: https://github.com/mdn/stumptown-content/issues/106
             return 'info_box-value';
         default:
-            throw new Error(`Unrecognized ingredient: ${ingredient}`);
+            throw new Error(`Unrecognized ingredient: ${ingredientName}`);
     }
 }
 

--- a/scripts/build-json/compose-attributes.js
+++ b/scripts/build-json/compose-attributes.js
@@ -56,7 +56,6 @@ async function packageAttribute(attributePath) {
 
     // extract the values property
     if (h2Headings.length === 2) {
-        valuesHeading = h2Headings[0];
         attribute.values = packageValues(dom);
     }
 

--- a/scripts/build-json/slice-prose.js
+++ b/scripts/build-json/slice-prose.js
@@ -1,5 +1,3 @@
-const fs = require("fs");
-
 const markdown = require("./markdown-converter");
 const jsdom = require("jsdom");
 
@@ -33,7 +31,7 @@ function extractFromSiblings(node, terminatorTag, contentType) {
     return content;
 }
 
-function getSectionValue(node, sections) {
+function getSectionValue(node) {
     const sectionName = node.textContent.trim();
     const sectionContent = extractFromSiblings(node, "H2", "html");
 
@@ -59,7 +57,7 @@ async function packageProse(proseMD) {
     let node = dom.firstChild;
     while (node) {
         if (node.nodeName === "H2") {
-            const sectionValue = getSectionValue(node, sections);
+            const sectionValue = getSectionValue(node);
             sections.push({
                 type: "prose",
                 value: sectionValue

